docs/source/development/overview.md

```diff
@@ -0,0 +1,319 @@
+get a feel for what they should do, but essentially unit tests should provide a set of
+known inputs to a function and check that the expected answer (which could be an
+Exception) is generated.
+
+Again, the `pytest` package and plugins are installed by `poetry`. See the [code testing
+page](./code_testing.md) for more details but you should be able to check the tests run
+using the following command. Be warned that the `mypy` steps can be very time consuming
+on the first run, but `pytest` does some cacheing that makes them quicker when they next
+run.
+
+```sh
+poetry run pytest
+```
+
+### Code profiling
+
+We use a specific set of `pytest` tests to carry out code profiling using
+the `pytest-profile` plugin. This makes it easier to spot functions that are running
+slowly within the `pyrealm` code. We also use our own `profiling/run-benchmarking.py`
+script to compare code profiles between versions to check that new code has not impacted
+performance.
+
+See the [profiling and benchmarking page](./profiling_and_benchmarking.md) for more
+details on running profiling.
+
+### The `pyrealm-build-data` package
+
+The `pyrealm` repository includes the [`pyrealm-build-data`
+package](./pyrealm_build_data.md), which is used to provide a range of datasets used in
+package testing and in documentation.
+
+### Documentation
+
+We use `sphinx` to maintain the documentation for `pyrealm` and Google style docstrings
+using the `napoleon` formatting to provide API documentation for the code. We use MyST
+Markdown to provide dynamically built usage examples. See the [documentation
+page](./documentation.md) for details.
+
+In order to build the documentation, you will need to register the `poetry` virtual
+enviroment, so that it can be used by `jupyter` and `myst` to build dynamic content.
+This only needs to be done once.
+
+```bash
+poetry run python -m ipykernel install --user --name=pyrealm_python3
+```
+
+After that, the following code can be used to build the documentation
+
+```bash
+# Build docs using sphinx
+cd docs
+poetry run sphinx-build -W --keep-going source build
+```
+
+### GitHub Actions
+
+We use GitHub Action workflows to update `pre-commit`, run code quality checks on pull
+requests, and to automate profiling and release publication. See the [GitHub Actions
+page](./github_actions.md) for details.
+
+### Package version releases
+
+We use trusted publishing from GitHub releases to release new versions of `pyrealm` to
+[PyPI](https://pypi.org/project/pyrealm/). Releases are also picked up and archived on
+[Zenodo](https://doi.org/10.5281/zenodo.8366847). See the [release process
+page](./release_process.md) for details.
+
+## Setup script example
+
+The scripts below bundle all the commands together to show the set up process, including
+using `pyenv` to mangage `python` versions.
+
+:::{admonition} Setup script
+
+``` sh
+#!/bin/bash
+
+# pyenv and poetry use sqlite3
+sudo apt install sqlite3 sqlite3-doc libsqlite3-dev
```

Comment:
I didn't find it was necessary to install these dependencies on my local install, but that might be because I used homebrew to install pyenv.